python/ray/data/_internal/metadata_exporter.py

```diff
@@ -120,12 +128,32 @@ class DatasetMetadata:
         topology: The structure of the dataset's operator DAG.
         dataset_id: The unique ID of the dataset.
         start_time: The timestamp when the dataset execution started.
+        data_context: The DataContext attached to the dataset.
     """
 
     job_id: str
     topology: Topology
     dataset_id: str
     start_time: float
+    data_context: DataContext
+
+
+def sanitize_for_struct(obj):
```

Comment:
actually should we just `json.dumps` the entire obj? 
this function seems to be just reinventing json.dumps.